### ISSUE TITLE
Demo code

### ISSUE THREAD

**Issue by JuewenPeng** [dulucas/Displacement_Field#10]
Hi, I write a demo code to do a test. The model is trained using the code in this repo. However, I find the visulization of the result is not as good as imagined. I was wondering if I did something wrong. Here are the demo code and testing images.

Code:
```
import numpy as np
import torch
import cv2
import matplotlib.pyplot as plt

from df import Displacement_Field


device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

model = Displacement_Field()
pth_path = 'xxx/Displacement_Field/log/nyu/df_nyu_rgb_guidance/snapshot/epoch-19.pth'
model.load_state_dict(torch.load(pth_path)['model'])
model.to(device)
model.eval()

image = cv2.imread('./image.png', cv2.IMREAD_GRAYSCALE)
depth = cv2.imread('./depth.png', cv2.IMREAD_GRAYSCALE)

image = torch.from_numpy(image.astype(np.float32) / 255.0).unsqueeze(0).unsqueeze(0).to(device)
depth = torch.from_numpy(depth.astype(np.float32) / 255.0).unsqueeze(0).unsqueeze(0).to(device)
depth = (depth - depth.min()) / (depth.max() - depth.min())

pred = model(image, depth)

pred = pred.detach().cpu().numpy()[0, 0]
plt.imsave('./result.png', pred, cmap='gray')
```

Image:
![image](https://user-images.githubusercontent.com/38718148/96842738-ff350900-147f-11eb-8e07-1a3687c735b0.png)

Depth:
![depth](https://user-images.githubusercontent.com/38718148/96842745-02c89000-1480-11eb-895a-b3487487e373.png)

Result:
![result](https://user-images.githubusercontent.com/38718148/96859424-1251d400-1494-11eb-86d2-737a2e6f5b3b.png)

**Comment by miangoleh** [comment]
I am trying to train the model but I am getting pretty much the same depth image as the input from the network. Were you able to fix the issue? 
When I look at the Loss logs they are not changing that much after the first epoch.

**Comment by JuewenPeng** [comment]
I am sorry but I can not fix it. Besides, I do more experiments and find that this code indeed improves the performance around the objects boundary in some degree. However, it also involve obvious artifacts in the final results.

**Comment by dulucas** [comment]
Hi, sorry for late response because I was focusing on another project. The model works fine for me on NYU dataset with the output of MDE models. For your issue, I think the main problem is that the blurry of your input depth images is not only caused by the MDE models, but also caused by image downsampling. We noticed that the two types of blurry are different and should be solved differently. As shown in your depth image, there is a lot of aliasing effects caused by image downsampling, some pixel values are missed during downsampling, you probably should first apply a image-superresolution network to obtain a depth image with higher resolution. Hope it helps.

**Comment by dulucas** [comment]
@mahdi139091 Could you please specify your problem ?

**Comment by dulucas** [comment]
@pengjw98 For your second question about the artifacts, lowring the learning rate works fine. The hyper-parameters in this repo may not be the optimal, as we have done our exps in a rush and focusing only on the numerical results. In addition to the learning rate, tuning the hyper-parameters in df.py(320, which is arbitrarily chosen) also will work I think. Hope it helps.

**Comment by dulucas** [comment]
@pengjw98 Also you can change the backbone network or the entire network architecture, the architecture that we adopt is very surely not the optimal one, because we just wanted to valid our idea with some simple networks instead of hacking the test dataset.

**Comment by JuewenPeng** [comment]
Thank you for your reply.

**Comment by dulucas** [comment]
@pengjw98 Hi, I pushed a new model which produce much less artifacts. Please refer to https://github.com/dulucas/Displacement_Field/commit/f918b54f32de3ebaac8937666e04db0012ab36c5 for more details.

**Comment by JuewenPeng** [comment]
That’s really nice. Thanks a lot.
